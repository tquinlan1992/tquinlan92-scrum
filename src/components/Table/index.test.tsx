import { TicketTable } from './';
import { shallow } from 'enzyme';
import * as React from 'react';

const ticket = {
    title: 'title',
    storyPoint: 3,
    description: 'description',
    deleted: true,
    closed: true,
    sprint: true,
    sprintName: 'sprintName',
    priority: 0,
    _id: '_id'
};

describe('test snapshot of component', () => {
    it('should render', () => {
        const props = {
            tickets: [
                ticket,
                {
                    ...ticket,
                    closed: false,
                },
                {
                    ...ticket,
                    closed: false,
                    
                }
            ],
            onClose: jest.fn(),
            onAddTicketToSprint: jest.fn(),
            onRemoveFromSprint: jest.fn(),
            title: 'title'
        };
        const result = shallow(<TicketTable {...props} />);
        expect(result).toMatchSnapshot();
    });
});
