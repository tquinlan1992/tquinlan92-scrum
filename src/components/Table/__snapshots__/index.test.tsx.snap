// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot of component should render 1`] = `
<Fragment>
  <WithStyles(Typography)
    gutterBottom={true}
    variant="h6"
  >
    title
  </WithStyles(Typography)>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          Ticket
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Sprint Name
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
        <WithStyles(TableCell) />
        <WithStyles(TableCell) />
        <WithStyles(TableCell) />
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        draggable={true}
        key="_id"
      >
        <WithStyles(TableCell)>
          <TextField
            defaultValue="title"
            required={false}
            select={false}
            variant="standard"
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          sprintName
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Closed
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Sprint
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Add To Sprint"
            variant="contained"
          >
             Add To Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Open Ticket"
            variant="contained"
          >
             Remove From Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Close Ticket"
            variant="contained"
          >
             Close Ticket 
          </WithStyles(Button)>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        draggable={true}
        key="_id"
      >
        <WithStyles(TableCell)>
          <TextField
            defaultValue="title"
            required={false}
            select={false}
            variant="standard"
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          sprintName
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
        <WithStyles(TableCell)>
          Sprint
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Add To Sprint"
            variant="contained"
          >
             Add To Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Open Ticket"
            variant="contained"
          >
             Remove From Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Close Ticket"
            variant="contained"
          >
             Close Ticket 
          </WithStyles(Button)>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        draggable={true}
        key="_id"
      >
        <WithStyles(TableCell)>
          <TextField
            defaultValue="title"
            required={false}
            select={false}
            variant="standard"
          />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          sprintName
        </WithStyles(TableCell)>
        <WithStyles(TableCell) />
        <WithStyles(TableCell)>
          Sprint
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Add To Sprint"
            variant="contained"
          >
             Add To Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Open Ticket"
            variant="contained"
          >
             Remove From Sprint 
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            title="Close Ticket"
            variant="contained"
          >
             Close Ticket 
          </WithStyles(Button)>
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</Fragment>
`;
